router/batchrouter/batchrouter.go

```diff
@@ -700,6 +700,7 @@ func (brt *HandleT) copyJobsToStorage(provider string, batchJobs *BatchJobsT, is
 			Config:           batchJobs.BatchDestination.Destination.Config,
 			UseRudderStorage: useRudderStorage,
 		}),
+		WorkspaceID: batchJobs.BatchDestination.Destination.WorkspaceID,
```

Comment:
Shouldn't we move the `WorkspaceID` field in `ObjectStorageOptsT` instead?
By doing it, `GetObjectStorageConfig` function would then be able to set the appropriate (new) entry in the `objectStorageConfigMap` if necessary for S3 and `GetS3Config` would simply use this new entry to populate `S3Config` with a new field, e.g. `ExternalID`.
Finally, `S3Manager#getSessionConfig` wouldn't need any `WorkspaceID` field, it would have all necessary information in its `S3Config`

Reply:
Implemented.